### PR TITLE
WSL automated installation script for programmers guide

### PR DESCRIPTION
**Pull request by SatyamRaina**
# Pull Request Description:

This pull request introduces a comprehensive script designed to automate the installation process of Windows Subsystem for Linux (WSL) along with essential dependencies required for development. The script includes the installation of SplashKit, .NET, Visual Studio Code, and language-specific tools to streamline the setup process for developers.

## Key Features:
- Automated installation of WSL, including necessary components and configurations.
- Installation of SplashKit for graphics and multimedia development.
- Integration of .NET framework for building cross-platform applications.
- Setup of Visual Studio Code with recommended extensions for a seamless coding experience.
- Inclusion of language tools for enhanced development capabilities.

This script aims to simplify the initial setup for developers by reducing manual steps and ensuring a consistent development environment across systems. Feedback and contributions are welcome to improve functionality and expand compatibility.

### CODE REVIEW

**Comment by hugglesfox** [comment]
Is this PR meant to change 133 files?

If so then it might be nice to split it into multiple commits. If that's not the intention and you need some help fixing the commit, feel free to message me on teams and we can sort it out

**Review by omckeon — CHANGES_REQUESTED** [review]
I've just tested this script on my Windows laptop. Everything works well except that we don't want to install VS Code in WSL - instead, the user will be using VS Code on Windows, and then will install the WSL extension to connect in with WSL.

File: scripts/demo.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+sudo apt update
+sudo apt-get install -y dotnet-sdk-8.0 dotnet-runtime-8.0
+echo 'export DOTNET_ROOT=$HOME/.dotnet' >> ~/.bashrc
+echo 'export PATH=$PATH:$HOME/.dotnet' >> ~/.bashrc
+
+# Install C++ tools
+echo "Installing C++ tools..."
+sudo apt-get -y install clang build-essential
+
+# Install VS Code
```
Comment (omckeon) [inline scripts/demo.sh:40]: Remove lines 40 - 45 (Not needed for WSL)

File: scripts/demo.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+wget -O vscode.deb https://go.microsoft.com/fwlink/?LinkID=760868
+sudo dpkg -i vscode.deb
+sudo apt-get install -f
+rm vscode.deb
+
+# Install VS Code extensions
+echo "Installing VS Code extensions..."
+code --install-extension ms-vscode.cpptools
+code --install-extension ms-dotnettools.csharp
+
```
Comment (omckeon) [inline scripts/demo.sh:51]: Missing a few extensions here. Update this part to be:

code --install-extension ms-vscode-remote.remote-wsl
code --install-extension ms-vscode.cpptools-extension-pack
code --install-extension ms-dotnettools.csharp
code --install-extension ms-dotnettools.csdevkit
code --install-extension ms-dotnettools.vscodeintellicode-csharp

**Comment by omckeon** [comment]
Can you make a new PR to add the updated script into a folder named 'wsl' in this repo: https://github.com/thoth-tech/skm/tree/develop/install-scripts. And name the file 'wsl-install.sh'.

**Comment by omckeon** [comment]
Closing this PR as it should just be for the script. The other changes should go into a separate PR.

**Pull request closed without merging**